wagtail_localize_rws_languagecloud/rws_client.py

```diff
@@ -159,18 +159,22 @@ def create_source_file(
         if not self.is_authenticated:
             raise NotAuthenticated()
 
+        # Assume the last three characters are ".po"
+        filename_sans_suffix = filename[:-3]
+        cleaned_filename = safe_characters.sub("", filename_sans_suffix) + ".po"
```

Comment:
https://github.com/zerolab/wagtail-localize-rws-languagecloud/blob/main/wagtail_localize_rws_languagecloud/sync.py#L240 indeed sends a ".po" file.

was going to nitpick that if someone uses this client to pass on a different file extension, then this would produce unintended results. However https://languagecloud.sdl.com/lc/api-docs/b3A6NzExNjgxMQ-add-source-file doesn't specify any restrictions in that regard, so `foo.txt` becoming `footxt.po` is not a big deal.

were I to nitpick, I'd have suggested something like

```
filename_parts = filename.split(os.extsep)
if len(filename_parts) > 1:
  ext = f"{os.extsep}.{filename_parts.pop()}"
else:
  ext = ""
filename_sans_suffix = "".join(filename_parts)
cleaned_filename = safe_characters.sub("", filename_sans_suffix) + ext
```

or something like that, but I am not going to 😂

Reply:
Nice! I think I will merge that in.